internal/pikamon/commands/catch.go

```diff
@@ -0,0 +1,123 @@
+package commands
+
+import (
+	"fmt"
+	"github.com/Jac0bDeal/pikamon/internal/pikamon/util"
+	"github.com/bwmarrin/discordgo"
+	"github.com/mtslzr/pokeapi-go"
+	log "github.com/sirupsen/logrus"
+	"math/rand"
+	"strconv"
+	"strings"
+)
+
+var pokemonExpiredMessages = []string{
+	"The pokemon heard you coming and ran for the hills!",
+	"The pokemon thought you were ugly and didn't want to be caught by an ugly person!",
+	"The pokemon likes Physics (mistakenly) and thought you were too dumb to understand it so it ran!",
+}
+
+var catchFailureMessages = []string{
+	"You may need to have your eyes checked!",
+	"Better luck next time!",
+	"Have you tried getting good?",
+}
+
+func pokemonExpiredFailure(s *discordgo.Session, m *discordgo.MessageCreate) bool {
+	expireMessageIndex := rand.Intn(len(pokemonExpiredMessages))
+	expireMessage := "You have failed to catch the pokemon! " + pokemonExpiredMessages[expireMessageIndex]
+	msg := discordgo.MessageEmbed{
+		Title:       "The Pokemon has run away!",
+		Description: expireMessage,
+	}
+	if _, err := s.ChannelMessageSendEmbed(m.ChannelID, &msg); err != nil {
+		log.Error(err)
+		return false
+	}
+	return true
+}
+
+func catchFailure(s *discordgo.Session, m *discordgo.MessageCreate) bool {
+	catchFailureMessageIndex := rand.Intn(len(catchFailureMessages))
+	expireMessage := fmt.Sprintf("%s has failed to catch the pokemon! %s", m.Author.Username, catchFailureMessages[catchFailureMessageIndex])
+	msg := discordgo.MessageEmbed{
+		Description: expireMessage,
+	}
+	if _, err := s.ChannelMessageSendEmbed(m.ChannelID, &msg); err != nil {
+		log.Error(err)
+		return false
+	}
+	return true
+}
+
+func publishSuccessfulCatch(s *discordgo.Session, m *discordgo.MessageCreate, pName string) bool {
+	// TODO - save to database
+	catchMessage := fmt.Sprintf("Congratulations %s! You caught a %s", m.Author.Username, pName)
+	msg := discordgo.MessageEmbed{
+		Description: catchMessage,
+	}
+	if _, err := s.ChannelMessageSendEmbed(m.ChannelID, &msg); err != nil {
+		log.Error(err)
+		return false
+	}
+	return true
+}
+
+func catch(s *discordgo.Session, m *discordgo.MessageCreate) {
+	// check if channel id is still in cache. If it is not then there is nothing to catch
+	p, exists := util.BotMetadata.ChannelCache.Get(m.ChannelID)
+	if !exists {
+		publishFailure := pokemonExpiredFailure(s, m)
```

Comment:
This should be a returned error, so we can log out what the error was

Reply:
done